tests/components/plugwise/test_init.py

```diff
@@ -151,3 +151,37 @@ async def test_migrate_unique_id_relay(
     entity_migrated = entity_registry.async_get(entity.entity_id)
     assert entity_migrated
     assert entity_migrated.unique_id == new_unique_id
+
+
+@pytest.mark.parametrize(
+    ("entitydata"),
+    [
+        {
+            "domain": SENSOR_DOMAIN,
+            "platform": DOMAIN,
+            "unique_id": f"{HEATER_ID}-indoor_temperature",
+            "suggested_object_id": "opentherm_indoor_temperature",
+            "disabled_by": None,
+        },
+    ],
+)
+async def test_entity_registry_cleanup(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_smile_anna: MagicMock,
+    entitydata: dict,
+) -> None:
+    """Test a clean-up of the entity_registry."""
+    mock_config_entry.add_to_hass(hass)
+
+    entity_registry = er.async_get(hass)
+    entity_registry.async_get_or_create(
+        **entitydata,
+        config_entry=mock_config_entry,
+    )
+    assert len(entity_registry.entities) == 1
+
+    assert await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert len(entity_registry.entities) == 24
```

Comment:
We should check the registry to see if the mocked device is removed instead of checking a total.

Reply:
Yes, sure, will update the test later today.

Reply:
Updated as suggested.